blocks/market-insight/market-insight.js

```diff
@@ -84,35 +56,53 @@ function addSocialButtonEvent(button, block) {
   });
 }
 
-async function decorateCards(block, placeholders, previousNode) {
-  const results = await fetchMarketInsightMockData();
+async function decorateCards(block, placeholders, cardCount, previousNode) {
+  const queryObj = await fetchMarketInsightMockData();
```

Comment:
Can we avoid await? Please see [this comment](https://github.com/aemsites/icicidirect/pull/82/files#r1548731220)

Reply:
We have implemented the `observe()` method, after the page loading, no cards are rendered until the end user scrolls to this block. I think what I need to do is add a loading icon or layer by default and remove it when the cards are loaded. What do you think?